src/tables.jl

```diff
@@ -126,25 +126,38 @@ uniquely identify rows in the `table` then an `ArgumentError` is throw. If `forc
 then the duplicate (non-unique) entries will be overwritten.
 """
 function populate!(A, table, value::Symbol; force=false)
-    # Use a BitArray mask to detect duplicates and error instead of overwriting.
-    mask = force ? falses() : falses(size(A))
-
-    for r in Tables.rows(table)
-        vals = Tuple(Tables.getcolumn(r, c) for c in dimnames(A))
-        inds = map(findindex, vals, axiskeys(A))
-
-        # Handle duplicate error checking if applicable
-        if !force
-            # Error if mask already set.
-            mask[inds...] && throw(ArgumentError("Key $vals is not unique"))
-            # Set mask, marking that we've set this index
-            setindex!(mask, true, inds...)
-        end
+    cols = Tables.columns(table)
```

Comment:
i wonder if we should can check how data is stored by checking [`Tables.columnaccess`](https://github.com/JuliaData/Tables.jl/blob/main/src/Tables.jl#L330-L336), and use that to decide whether to call `Tables.columns` or `Tables.rows` (i.e. the old implementation) 

e.g.
```
if Tables.columnaccess(table)
     cols = Tables.colums(table)
     ....
else
    for r in Tables.rows(table)
    ...
end
```
(we'd have to benchmark if that helps on row tables or not)